### PR TITLE
Live 81 get user info from backend

### PR DESCRIPTION
**Pull request by Karolk99**
closes: [ #66 ](https://membraneframework.atlassian.net/jira/software/c/projects/LIVE/boards/11?modal=detail&selectedIssue=LIVE-66)

### CODE REVIEW

File: lib/membrane_live_web/controllers/webinar_controller.ex [hunk 1]
```diff
@@ -14,7 +14,8 @@ defmodule MembraneLiveWeb.WebinarController do
 
   @spec create(any, map) :: any
   def create(conn, %{"webinar" => webinar_params}) do
-    with {:ok, %Webinar{} = webinar} <- Webinars.create_webinar(webinar_params) do
+    with {:ok, %Webinar{} = webinar} <-
+           Webinars.create_webinar(webinar_params, conn.assigns.user_id) do
```
Comment (Rados13) [inline lib/membrane_live_web/controllers/webinar_controller.ex:18]: Why do we need here with, as we don't even handle errors?

File: test/membrane_live/webinars_test.exs [hunk 2]
```diff
@@ -33,15 +34,17 @@ defmodule MembraneLive.WebinarsTest do
         "title" => "some title"
       }
 
-      assert {:ok, %Webinar{} = webinar} = Webinars.create_webinar(valid_attrs)
+      user = user_fixture()
```
Comment (Rados13) [inline test/membrane_live/webinars_test.exs:37]: As we call this `user_fixture` multiple time we could move this to constant or setup.

File: test/membrane_live_web/channels/event_channel_test.exs [hunk 3]
```diff
@@ -16,73 +17,54 @@ defmodule MembraneLiveWeb.EventChannelTest do
...
-        })
+    test "check if users are in presence", %{user: user, uuid: uuid, pub_socket: pub_socket} do
+      socket2 = MembraneLiveWeb.EventSocket |> socket("event_id1", %{})
 
-      assert return_value == {:error, %{reason: "Viewer with this name already exists."}}
-    end
+      google_claims = %{
+        "name" => "mock_user2",
+        "email" => "mock_email2@gmail.com",
+        "picture" => "https://google.com/2"
```
Comment (Rados13) [inline test/membrane_live_web/channels/event_channel_test.exs:50]: Couldn't we first `create_user` and then use `create_channel_connection`?

File: test/membrane_live_web/channels/event_channel_test.exs [hunk 4]
```diff
@@ -163,4 +145,18 @@ defmodule MembraneLiveWeb.EventChannelTest do
...
+    {:ok, user, token}
+  end
+
+  defp create_channel_connection(uuid, token) do
+    MembraneLiveWeb.EventSocket
+    |> socket("event_id", %{})
+    |> subscribe_and_join(MembraneLiveWeb.EventChannel, "event:#{uuid}", %{
+      token: token
+    })
+  end
```
Comment (Rados13) [inline test/membrane_live_web/channels/event_channel_test.exs:161]: As other tests use this function you can move it to the `testUtills` module, so you won't duplicate code.

File: assets/js/pages/Auth.tsx [hunk 5]
```diff
@@ -3,15 +3,20 @@ import { useNavigate } from "react-router-dom";
...
+} from "../utils/storageUtils";
 const Auth = () => {
   const navigate = useNavigate();
   const redirectToHomePage = () => navigate("/");
 
   const fetchToken = async (googleResponse) => {
     try {
       const response = await axios.post("auth", googleResponse);
-      localStorage.setItem("jwt", response.data.token);
+      storageSetToken(response.data.token);
```
Comment (Rados13) [inline assets/js/pages/Auth.tsx:19]: Shouldn't we check, if this token is defined?

File: assets/js/pages/Auth.tsx [hunk 6]
```diff
@@ -20,9 +25,18 @@ const Auth = () => {
...
     await fetchToken(response);
-    console.log(isUserAuthenticated());
     if (isUserAuthenticated()) {
-      redirectToHomePage();
+      axios
+        .get("/me")
+        .then((response) => {
+          storageSetName(response.data.name);
+          storageSetEmail(response.data.email);
+          storageSetPicture(response.data.picture);
```
Comment (Rados13) [inline assets/js/pages/Auth.tsx:34]: As above shouldn't we check, if this exists?

File: assets/js/utils/storageUtils.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+export const storageGetName = (): string => {
+  const name: string | null = localStorage.getItem("name");
```
Comment (Rados13) [inline assets/js/utils/storageUtils.ts:2]: You could move this name to variable.

File: assets/js/utils/storageUtils.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  localStorage.setItem("email", email);
+};
+
+export const storageSetPicture = (picture: string): void => {
+  localStorage.setItem("picture", picture);
+};
+
+export const storageSetToken = (token: string): void => {
+  localStorage.setItem("jwt", token);
+};
```
Comment (Rados13) [inline assets/js/utils/storageUtils.ts:24]: As above.

**Review by Rados13 — CHANGES_REQUESTED** [review]

File: test/membrane_live_web/channels/event_channel_test.exs [hunk 9]
```diff
@@ -16,73 +18,49 @@ defmodule MembraneLiveWeb.EventChannelTest do
         title: "Test webinar"
       })
```
Comment (Rados13) [inline test/membrane_live_web/channels/event_channel_test.exs:19]: IMO this shouldn't be possible to add a Webinar without a moderator.

**Review by Rados13 — CHANGES_REQUESTED** [review]

File: test/membrane_live_web/channels/event_channel_test.exs (hunk 9 above)
Comment (Karolk99) [reply test/membrane_live_web/channels/event_channel_test.exs:19]: Will be solved in https://membraneframework.atlassian.net/jira/software/c/projects/LIVE/boards/11?modal=detail&selectedIssue=LIVE-94

**Review by Rados13 — COMMENTED** [review]
There are some merge conflicts.

File: assets/js/pages/Auth.tsx [hunk 10]
```diff
@@ -11,7 +16,9 @@ const Auth = () => {
   const fetchToken = async (googleResponse) => {
     try {
       const response = await axiosWithInterceptor.post("auth", googleResponse);
-      setJwt(response.data);
+      if (response.data.authToken === undefined || response.data.refreshToken === undefined)
```
Comment (Rados13) [inline assets/js/pages/Auth.tsx:19]: Do we need this `=== undefined`?

**Review by Rados13 — APPROVED** [review]

**Pull request merged**